pkg/artifacts/signable.go

```diff
@@ -257,6 +268,104 @@ func extractTargetFromResults(obj objects.TektonObject, identifierSuffix string,
 	return ss
 }
 
+// ExtractStructuredTargetFromResults extracts structured signable targets aim to generate intoto provenance as materials within TaskRun results and store them as StructuredSignable.
+// categoryMarker can either be ARTIFACT_INPUTS or ARTIFACT_OUTPUTS.
+func ExtractStructuredTargetFromResults(obj objects.TektonObject, categoryMarker string, logger *zap.SugaredLogger) []*StructuredSignable {
+	objs := []*StructuredSignable{}
+	for _, res := range obj.GetResults() {
+		if strings.HasSuffix(res.Name, categoryMarker) {
+			if res.Value.ObjectVal["uri"] == "" {
+				logger.Errorf("%s should have uri key", res.Name)
+				continue
+			}
+			if res.Value.ObjectVal["digest"] == "" {
+				logger.Errorf("%s should follow digest key", res.Name)
+				continue
+			}
+			if err := checkDigest(res.Value.ObjectVal["digest"]); err != nil {
+				logger.Errorf("error getting digest %s: %v", res.Value.ObjectVal["digest"], err)
+				continue
+			}
+			if res.Value.ObjectVal["uri"] == "" {
+				logger.Errorf("URI cannot be empty for %s", res.Name)
+				continue
+			}
+			objs = append(objs, &StructuredSignable{URI: res.Value.ObjectVal["uri"], Digest: res.Value.ObjectVal["digest"]})
+		}
+	}
+	return objs
+}
+
+// MergeStructuredesultsIntoMaterials merge structure.
+func MergeStructuredResultsIntoMaterials(mats *[]slsa.ProvenanceMaterial, obj objects.TektonObject, categoryMarker string, logger *zap.SugaredLogger) {
+	// Retrieve structured provenance for inputs.
+	ssts := extractStructuredTargetFromResults(obj, ArtifactsInputsResultName, logger)
+	for _, s := range ssts {
+		splits := strings.Split(s.Digest, ":")
+		alg := splits[0]
+		digest := splits[1]
+		*mats = append(*mats, slsa.ProvenanceMaterial{
+			URI:    s.URI,
+			Digest: map[string]string{alg: digest},
+		})
+	}
+}
+
+// extractStructuredTargetFromResults extracts structured signable targets aim to generate intoto provenance as materials within TaskRun results and store them as StructuredSignable.
+// categoryMarker categorizes signable targets into inputs and outputs.
+func extractStructuredTargetFromResults(obj objects.TektonObject, categoryMarker string, logger *zap.SugaredLogger) []*StructuredSignable {
+	objs := []*StructuredSignable{}
+	// TODO(#592): support structured results using Run
+	results := []objects.Result{}
+	tr, isTaskRun := obj.GetObject().(*v1beta1.TaskRun)
+	if isTaskRun {
```

Comment:
Or are you referring to something else here?

Reply:
Marking this as resolved. Feel free to re-open this if you see anything else needs to be addressed here

Reply:
Hmmm that looks like an oversight (especially since [TaskRuns seem to support it](https://github.com/tektoncd/chains/blob/780521656ab0ea2a72d7a969a13040848ac6b969/pkg/chains/objects/objects.go#L98-L99)). Can you open up an issue? I think I'd lean towards adding it to the object types long term.

cc @bcaton85 @lcarva fyi (I think this also affects array results)

Reply:
Created #595 for tracking that